sdk/identity/identity/review/identity-api-review.md

```diff
@@ -0,0 +1,137 @@
+## API Report File for "@azure/identity"
+
+> Do not edit this file. It is a report generated by [API Extractor](https://api-extractor.com/).
+
+```ts
+
+import { AccessToken } from '@azure/core-http';
+import { GetTokenOptions } from '@azure/core-http';
+import { ServiceClientOptions } from '@azure/core-http';
+import { TokenCredential } from '@azure/core-http';
+
+export { AccessToken }
+
+// @public
+export class AggregateAuthenticationError extends Error {
+    constructor(errors: any[]);
+    errors: any[];
+}
+
+// @public
+export const AggregateAuthenticationErrorName = "AggregateAuthenticationError";
+
+// @public
+export class AuthenticationError extends Error {
+    constructor(statusCode: number, errorBody: object | string | undefined | null);
+    readonly errorResponse: ErrorResponse;
+    readonly statusCode: number;
+}
+
+// @public
+export const AuthenticationErrorName = "AuthenticationError";
+
+// @public
+export class AuthorizationCodeCredential implements TokenCredential {
+    constructor(tenantId: string, clientId: string, clientSecret: string | undefined, authorizationCode: string, redirectUri: string, options?: IdentityClientOptions);
+    getToken(scopes: string | string[], options?: GetTokenOptions): Promise<AccessToken | null>;
+    }
+
+// @public
+export type BrowserLoginStyle = "redirect" | "popup";
+
+// @public
+export class ChainedTokenCredential implements TokenCredential {
+    constructor(...sources: TokenCredential[]);
+    getToken(scopes: string | string[], options?: GetTokenOptions): Promise<AccessToken | null>;
```

Comment:
The '| null' seems a bit out of place here (authentication feels very much like it either works or it doesn't).

What's the use case for null? Should the user retry?

Reply:
Agreed. Was also wondering if in general we should be preferring `undefined` over `null` in cases where we don't have a shape to return. The event-hubs send method returns undefined for example, and I've gotten feedback to use `undefined` over null in internal APIs.

Reply:
Could/should it reject instead of resolving to `null` in the event that none of the chained token providers is able to resolve a token? The source seems to suggest that it cannot resolve to `null` unless no errors are produced in the chain.

Reply:
leave as null sentinal "value"

Reply:
same as above